src/argilla/server/search_engine.py

```diff
@@ -139,6 +151,90 @@ async def add_records(self, dataset: Dataset, records: Iterable[Record]):
         if errors:
             raise RuntimeError(errors)
 
+    async def search(
+        self,
+        dataset: Dataset,
+        query: Union[Query, str],
+        user_response_status_filter: Optional[UserResponseStatusFilter] = None,
+        limit: int = 100,
+    ) -> SearchResponses:
+        # See https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html
+
+        if isinstance(query, str):
+            query = Query(text=TextFieldQuery(q=query))
+
+        bool_query = {"must": []}
+        if not query.text.field:
+            # https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-combined-fields-query.html
+            text_query = {
+                "combined_fields": {
+                    "query": query.text.q,
+                    "fields": [f"fields.{field.name}" for field in dataset.fields],
+                    "operator": "and",
+                }
+            }
```

Comment:
Maybe we could move this to a function returning `text_query` as it would be easier to test with unit tests.